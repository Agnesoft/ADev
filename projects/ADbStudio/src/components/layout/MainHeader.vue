<template>
    
<header class="page-header">
    <router-link :to="`/${$i18n.locale}`" class="logo">ADbStudio</router-link>
</header>
  
</template>

<script>
export default {
    name: "MainHeader"
}
</script>

<style scoped>
    .page-header {
        --button-size: 2.3rem;
        display: grid;
        padding: 20px;
        gap: 20px;
        grid-template-columns: minmax(5rem,1fr);
        grid-template-areas:
            'logo';
        background-color: white;
        padding: 1rem 1rem 0 1rem;
        border-bottom: 1px solid var(--light-primary-color);
        box-shadow: 0 2px 4px rgba(0,0,0,.0622169);
    }
    .logo{
        color: var(--secondary-color);
        font-size: 2rem;
        font-weight: 700;
        grid-area: logo;
        font-family: var(--primary-font);
    }
    .logo:hover{
        opacity: 0.7;
    }
</style>
