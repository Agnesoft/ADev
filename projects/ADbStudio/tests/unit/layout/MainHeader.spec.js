import { shallowMount  } from '@vue/test-utils';
import MainHeader from '@/components/layout/MainHeader.vue';
//import LanguageSwitcher from "@/components/layout/LanguageSwitcher";

import { localVue, i18n } from '../TestUtils'


describe('MainHeader',() => {
    let wrapper
    beforeAll(() => {
        wrapper = shallowMount(MainHeader, {
            localVue,
            i18n,
          });
    })
    it('renders title and components', () => {
        //expect(wrapper.findComponent(LanguageSwitcher).exists()).toBe(true);
        expect(wrapper.find("header").exists()).toBe(true);
        expect(wrapper.find(".logo").text()).toBe("ADbStudio");
    })
})