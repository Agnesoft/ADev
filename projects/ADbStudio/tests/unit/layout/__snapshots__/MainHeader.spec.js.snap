// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders correctly (snapshot) 1`] = `
<header
  class="page-header"
>
  <router-link-stub
    ariacurrentvalue="page"
    class="logo"
    event="click"
    tag="a"
    to="/en"
  >
    ADbStudio
  </router-link-stub>
</header>
`;
